lightning/src/ln/channelmanager.rs

```diff
@@ -3754,8 +3760,9 @@ impl<'a, ChanSigner: ChannelKeys + Readable, M: Deref, T: Deref, K: Deref, F: De
 			}
 		}
 
+		const MAX_ALLOC_SIZE: u64 = 1024 * 64;
 		let forward_htlcs_count: u64 = Readable::read(reader)?;
-		let mut forward_htlcs = HashMap::with_capacity(cmp::min(forward_htlcs_count as usize, 128));
+		let mut forward_htlcs = HashMap::with_capacity(cmp::min(forward_htlcs_count as usize, (MAX_ALLOC_SIZE as usize)/mem::size_of::<HTLCForwardInfo>()));
```

Comment:
Oops, the inner in the hashmap isn't a HTLCForwardInfo, its a (u64, Vec<_>). The HTLCForwardInfo-relevant line is four lines down when we allocate the Vec::with_capacity(). The same goes for claimable_htlcs below.

Reply:
Addressed `forward_htlcs` and `claimable_htlcs` (put them back to 128) and for `per_peer_state`'s size, used its type: `<RwLock<HashMap<PublicKey, Mutex<PeerState>>>>` for the denominator under MAX_ALLOC_SIZE

Reply:
Instead of `RwLock<HashMap<PublicKey, Mutex<PeerState>>>`, you probably want `(PublicKey, Mutex<PeerState>)`.

The goal here is to capture that we only ever want to allocate up to 64KB for internal contents at a time - in a Vec the internal, heap-allocated things are of type X (where the vec is Vec<X>). In a HashMap, the internal contents are usually (K, V) (where its a HashMap<K, V>), though I don't think Rust guarantees this, its close enough for our purposes, it doesn't have to be exact. We always allocate the RwLock<HashMap<>> data on the stack, so its not the variable part, its the number of (K, V) entries that is the variable part.